daffodil-tdml/src/main/scala/org/apache/daffodil/tdml/TDMLException.scala

```diff
@@ -17,16 +17,72 @@
 
 package org.apache.daffodil.tdml
 
-class TDMLException(msg: String, val causes: Seq[Throwable])
-  extends Exception(msg, if (causes.length > 0) causes(0) else null) {
+import org.apache.daffodil.api.Diagnostic
+import org.apache.daffodil.util.Maybe
+import org.junit.AssumptionViolatedException
 
-  def this(msg: String) = this(msg, Nil)
+object TDMLException {
 
-  def this(cause: Throwable) = this(cause.getMessage(), List(cause))
+  def msgWithImpl(msg: String, implementation: String) =
+    if (implementation != "none" &&
```

Comment:
Changing to an Option type makes sense. Optional arg would just get left off i think. Making it required arg is important since you really do need to consider whether the exception is implementation-specific or not.